lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2538,6 +2552,31 @@ auto ClangImporter::Implementation::importFullName(
 
   // Omit needless words.
   StringScratchSpace omitNeedlessWordsScratch;
+
+  // swift_newtype-ed declarations may have common words with the type name
+  // stripped.
+  if (auto newtypeDecl = findSwiftNewtype(D)) {
+    SmallString<32> newNameBuffer;
+    auto newtypeNameWords = camel_case::getWords(newtypeDecl->getName());
+    auto declNameWords = camel_case::getWords(baseName);
+
+    // Scan the decl name, and drop any words that appear in order in the type
+    // name
+    for (auto wI = declNameWords.begin(), skipWI = newtypeNameWords.begin();
```

Comment:
Is there truly no utility within the enum-handling code to compare common prefixes between two names?